### ISSUE TITLE
markup Menus customizing items and order

### ISSUE THREAD

**Issue by kohsah** [cirsfid-unibo/lime#17]
I have been trying to understand how the order of items shown in the markup menu can be modified, and how certain  items can be removed and others added. 

I looked at markupMenu.json for the type  I am trying to customize, but there is no real order specified there, but in markupMenu_rules.json : 

```
{
    "rootElements": ["coverPage", "preface", "mainBody", "conclusions", "attachments", "components"],
    "commonElements": ["commonElement", "hierarchicalStructure",  "genericElement", "externalFragment", "commonsInline", "contentInline", "commonReference", "commonReport", "remark",  "foreign", "speech", "narrative", "debateBody"],
```

Is that where I can customize the order ?

**Comment by kohsah** [comment]
OK figured out that the root level markup menus can be reordered by editing rootElements in "markupMenu_rules.json"

But how to do the same for items within `preface` for instance which is one level within  ?

![screenshot from 2016-03-03 15 12 21](https://cloud.githubusercontent.com/assets/5685392/13490433/8cc9f9a8-e152-11e5-9a99-12635548110f.png)

**Comment by obujor** [comment]
Yes the markupMenu_rules.json is the right place to specify the order of the buttons.  For the order inside the elements markup menu keeps the order of the "children" list specified in the same file.
If you want to remove a button from the markup menu you have to remove the element from the list which contains that element from markupMenu_rules.json.

Pay attention on which markupMenu_rules.json file you are going to edit, there are many markupMenu_rules.json files and some of them are merged together. 
For example:
'/languagesPlugins/akoma3.0/interface/doc/markupMenu_rules.json'
is merged with the default file in
'/languagesPlugins/akoma3.0/interface/default/markupMenu_rules.json'
In the file inside "doc" folder you can change the configuration inside the "default" rules. 
So for example if you want to have only the 'p' element in the preface in the 'doc' documents you need to add in the '/languagesPlugins/akoma3.0/interface/doc/markupMenu_rules.json' file in the "elements" object the following element:
`"preface": { "children": ["p" ]}`
this overwrites the configuration of "preface" specified in the "default" markup rules.

In conclusion if in your markupMenu_rules.json file you don't find an element inside the "elements" object thas means that its configuration is taken from the "default" markup rules file.

**Comment by kohsah** [comment]
Thanks @obujor , hadn't noticed the default configuration, got it now .
